src/newwave/entity.mo

```diff
@@ -17,88 +17,275 @@ import Result "mo:base/Result";
 import Random "mo:base/Random";
 import Blob "mo:base/Blob";
 
-import EntityType "entity_type";
-import EntitySettings "entity_settings";
-
 import Utils "Utils";
-
-// example to include functions on type object:
-  // type Counter = { inc : () -> Nat };
-  /* func Counter() : { inc : () -> Nat } =
-  object {
-    var c = 0;
-    public func inc() : Nat { c += 1; c }
-  }; */
+import Text "mo:base/Text";
+import Types "Types";
 
 module {
 
+  /**
+   * Types of errors for finding the attached bridges
+  */
+  public type EntityAttachedBridgesErrors = {
+    #EntityNotFound;
+    #Error;
+  };
+
+  /**
+   * Return type for when finding the ids of attached bridges
+  */
+  public type EntityAttachedBridges = Types.Result<[Text], EntityAttachedBridgesErrors>;
```

Comment:
maybe call it something like EntityAttachedBridgesResult and define a type for the success case (could be EntityAttachedBridges) that in this case would be [Text] (but might change later)

Reply:
Good call, will change it